src/index.js

```diff
@@ -3,22 +3,52 @@ import { createServer } from '~/src/server'
 import { createLogger } from '~/src/server/common/helpers/logging/logger'
 
 const logger = createLogger()
+let server
 
 process.on('unhandledRejection', (error) => {
   logger.info('Unhandled rejection')
   logger.error(error)
   process.exit(1)
 })
 
+process.on('SIGINT', () => {
+  stopServer(server)
+})
+
+process.on('SIGTERM', () => {
+  stopServer(server)
+})
+
+startServer().catch((error) => {
+  logger.info('Server failed to start :(')
+  logger.error(error)
+})
+
 async function startServer() {
-  const server = await createServer()
+  server = await createServer()
   await server.start()
 
   server.logger.info('Server started successfully')
   server.logger.info(`Access your frontend on ${config.get('appBaseUrl')}`)
 }
 
-startServer().catch((error) => {
-  logger.info('Server failed to start :(')
-  logger.error(error)
-})
+function stopServer(server) {
```

Comment:
this arg is being shadowed - https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=457&id=DEFRA_cdp-portal-frontend